app/routes/analyzer.tsx

```diff
@@ -735,6 +751,88 @@ function AbilityPointsDetails({
   );
 }
 
+/**
+ * From an array of Main abilities, create a map of <Ability, number>, then return it as an Array after sorting by value, descending.
+ *
+ * The map describes the number of Ability chunks required to replace the main ability on a piece of gear.
+ *
+ * Extra processing is required for Main abilities that are primary slot-only abilities,
+ *  as they are comprised of 3 stackable ability chunks at a lower ability chunk count than usual.
+ */
+function getAbilityChunksMapAsArray(mainAbilities: AbilityWithUnknown[]) {
+  const abilityChunksMap: MainAbilityChunks = new Map();
+  for (let i = 0; i < mainAbilities.length; i++) {
+    const mainAbility = mainAbilities[i];
+    if (typeof mainAbility !== "undefined" && mainAbility !== "UNKNOWN") {
+      const primarySlotOnlyAbilityRef = abilities.filter(
+        (a) => a.name === mainAbility && a.abilityChunkTypesRequired.length > 0
+      );
+
+      // Primary-slot-only item that can have ability chunks put on it
+      if (primarySlotOnlyAbilityRef.length === 1) {
+        const primaryAbility = primarySlotOnlyAbilityRef[0];
+        if (typeof primaryAbility !== "undefined") {
+          for (
+            let j = 0;
+            j < primaryAbility?.abilityChunkTypesRequired.length;
+            j++
```

Comment:
Do this instead:

```suggestion
          for (const ability of primaryAbility.abilityChunkTypesRequired)
```

Also: if you have write `j < primaryAbility?.abilityChunkTypesRequired.length` something went wrong since you shouldn't be comparing number and `undefined`